### PR TITLE
Doc: Minor and not so minor RM edits

### PR DESCRIPTION
**Pull request by davidcok**

### CODE REVIEW

File: RELEASE.txt [hunk 1]
```diff
@@ -29,4 +29,72 @@ the release notes. For example:
 7) Add all the .zip files created by step 3 in the Package folder.
 
 8) Publish the release (by clicking the button at the bottom of the page)
-   and announce it to the world.
+
+9) Uodate the brew formula for dafny (see below)
```
Comment (RustanLeino) [inline RELEASE.txt:33]:
```suggestion
9) Update the brew formula for dafny (see below)
```

File: docs/_includes/Expressions.md [hunk 2]
```diff
@@ -466,21 +475,20 @@ TO BE WRITTEN - argument that describes how to initialize the array
 
 ## Object Allocation
 ````grammar
-ObjectAllocation_ = "new" Type [ "(" [ Expressions ] ")" ]
+ObjectAllocation_ = "new" Type [ "." ( Ident | digits ) ]
+                               [ "(" [ Expressions ] ")" ]
 ````
 
 This allocated a new object of a class type as explained
```
Comment (RustanLeino) [inline docs/_includes/Expressions.md:482]:
```suggestion
This allocates a new object of a class type as explained
```

File: docs/_includes/Introduction.md [hunk 3]
```diff
@@ -20,29 +20,33 @@ recursive functions, and types like sets and sequences. Specifications
...
 The easiest way to try out [Dafny is in your web browser at
 rise4fun](http://rise4fun.com/Dafny)[@Rise4fun:dafny]. Once you get a bit
-more serious, you may prefer to [download](http://dafny.codeplex.com/) it
+more serious, you may prefer to [download](https://github.com/dafny-lang/dafny/releases) it
 to run it on your machine. Dafny can be run from the command
-line (on Windows or other platforms) or from an IDE
+line (on Linux, MacOS, Windows or other platforms) or from an IDE
 such as emacs or
-Microsoft Visual Studio 2012 (or newer), where the Dafny
+Microsoft Visual Studio 2012 (or newer), where the `dafny`
```
Comment (RustanLeino) [inline docs/_includes/Introduction.md:38]: Let's not mention Visual Studio. We no longer provide the Dafny extension for Visual Studio.

File: docs/_includes/Types.md [hunk 4]
```diff
@@ -2957,16 +2952,17 @@ known constants at compile-time. They need not be numeric literals; combinations
...
 in [Section: Compile-Time Constants](#sec-compile-time-constants).
 
-## Numeric conversion operations
+## Conversion operations {#sec-conversion}
 
-For every numeric type `N`, there is a conversion function with the
-name `as N`.  It is a partial identity function.  It is defined when the
-given value, which can be of any numeric type, is a member of the type
+For every type `N`, there is a conversion operation with the
+name `as N`, described more fully in [a later section](#sec-as-expression).
```
Comment (RustanLeino) [inline docs/_includes/Types.md:2958]: This is not yet supported for all types. Is this sentence perhaps forward-looking in the same way as the description of `is` (see L2991)?

File: docs/_includes/Types.md [hunk 5]
```diff
@@ -2984,8 +2980,15 @@ If the compiler does specialize the run-time representation for
 respectively three, run-time conversions.
 
 The `as N` conversion operation is grammatically a suffix operation like
-`.`field and array indexing. Thus the `as` operation binds more tightly than
-prefix or binary operations: `- x as int` is `- (x as int)`; `a + b as int` is `a + (b as int)`.
+`.`field and array indexingi, but binds less tightly than unary operations:
```
Comment (RustanLeino) [inline docs/_includes/Types.md:2983]:
```suggestion
`.`field and array indexing, but binds less tightly than unary operations:
```

File: docs/_includes/Types.md [hunk 6]
```diff
@@ -2984,8 +2980,15 @@ If the compiler does specialize the run-time representation for
...
+`- x as int` is `(- x) as int`; `a + b as int` is `a + (b as int)`.
+
+There is also a corresponding [`is` operation](#sec-as-expression) that
+tests whether a value is valid for a given type. For example, `-5 as nat` is
```
Comment (RustanLeino) [inline docs/_includes/Types.md:2987]:
```suggestion
tests whether a value is valid for a given type. For example, `-5 is nat` is
```

**Review by RustanLeino — CHANGES_REQUESTED** [review]

File: RELEASE.txt (hunk 1 above)
Comment (davidcok) [reply RELEASE.txt:33]: Done

File: docs/_includes/Introduction.md (hunk 3 above)
Comment (davidcok) [reply docs/_includes/Introduction.md:38]: OK

File: docs/_includes/Types.md (hunk 4 above)
Comment (davidcok) [reply docs/_includes/Types.md:2958]: Yes -- and I mention just a bit later that this is only implemented for numeric base types so far.

File: docs/_includes/Types.md (hunk 5 above)
Comment (davidcok) [reply docs/_includes/Types.md:2983]: OK

File: docs/_includes/Types.md (hunk 6 above)
Comment (davidcok) [reply docs/_includes/Types.md:2987]: Aaergh. OK.

**Pull request merged**